src/VersionParser.php

```diff
@@ -41,6 +41,9 @@ class VersionParser
     /** @var array */
     private static $stabilities = array('stable', 'RC', 'beta', 'alpha', 'dev');
 
+    /** @var array  */
+    private static $constraintCache = array();
```

Comment:
This is already done in https://github.com/composer/composer/blob/master/src/Composer/Package/Version/VersionParser.php#L30-L32 so I doubt it's necessary/helpful here?

The one difference is it only caches parseConstraint**s** and not individual constraint. Is the gain of doing that really that big?

Reply:
I noticed that it were done in composer too, but think it's more efficient here and could then been removed from `composer/composer`.

Yes caching individual constraint have a significant gain `^1.0`, `^1.0 || ^2.0` will instantiate 2 Constraints (compared to 3 without cache). 
Which also have performance impact where `match` result is also cached